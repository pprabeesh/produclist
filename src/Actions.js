export const actions = {
  add: products => ({
    type: "add",
    productList: products
  }),
  update: filterBy => ({
    type: "update",
    data: filterBy
  }),
  filter: filterOption => ({
    type: "filter",
    selectedOption: filterOption
  })
};
