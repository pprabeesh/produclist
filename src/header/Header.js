import React from "react";
import { StyledHeader, StyledTitle, StyledFilter } from "./Header.styles";
const Header = () => (
  <StyledHeader>
    <StyledTitle> Woman's tops </StyledTitle>
    <StyledFilter>
      <select>
        <option>Filter by size</option>
        <option>Filter by price</option>
      </select>
    </StyledFilter>
  </StyledHeader>
);

export default Header;
