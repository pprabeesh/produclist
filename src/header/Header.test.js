import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { actions } from "../Actions";
import Header from "./Header";
import { FilterContext } from "../Context";

test("Header renders correctly", () => {
  const state = {};
  const dispatch = () => {};
  let container = document.createElement("div");

  renderer.act(() => {
    ReactDOM.render(
      <FilterContext.Provider value={{ state, actions, dispatch }}>
        <Header />
      </FilterContext.Provider>,
      container
    );
  });

  expect(container).toMatchSnapshot("Not expanded");
});
