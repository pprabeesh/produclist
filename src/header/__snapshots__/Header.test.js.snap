// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly: Not expanded 1`] = `
<div>
  <div
    class="sc-bwzfXH sc-htpNat bHldxh sc-bdVaJa iHZvIS"
  >
    <div
      class="sc-bdVaJa sc-bxivhb eufOPG"
    >
       Woman's tops 
    </div>
    <div
      class="sc-bdVaJa sc-ifAKCX fgROdP"
    >
      <select>
        <option
          value=""
        >
          Select
        </option>
        <option
          value="size"
        >
          Filter by size
        </option>
        <option
          value="price"
        >
          Filter by price
        </option>
      </select>
    </div>
  </div>
</div>
`;
